esrally/log.py

```diff
@@ -43,6 +44,44 @@ def log_config_path():
     return os.path.join(paths.rally_confdir(), "logging.json")
 
 
+def add_missing_loggers_to_config():
+    """
+    Ensures that any missing top level loggers in resources/logging.json are
+    appended to an existing log configuration
+    """
+    logger = logging.getLogger(__name__)
+
+    def missing_keys(source, target):
+        """
+        Returns any top-level dicts present in 'source', but not in 'target'
+        :return: A dict of all keys present in 'source', but not in 'target'
+        """
+        missing_keys = {}
+        for k in source:
+            if k in source and k in target:
+                continue
+            else:
+                missing_keys[k] = source[k]
+        return missing_keys
+
+    source_path = io.normalize_path(os.path.join(os.path.dirname(__file__), "resources", "logging.json"))
+    with open(log_config_path(), "r+", encoding="UTF-8") as target:
+        with open(source_path, "r+", encoding="UTF-8") as src:
```

Comment:
No need for write here:

```suggestion
        with open(source_path, "r", encoding="UTF-8") as src:
```